stats.go

```diff
@@ -129,6 +129,7 @@ func (sd *StatsDumper) dumpNow() {
 
 	if invalid > 0 {
 		m := make(map[string]int64)
+		t.mu.Lock()
```

Comment:
Is not sufficient the `RLock` here?

Reply:
This lock is taken once a second so it would probably not change anything to complicate the semantics here.
I don't even know why this is a rwmutex, it should be a simple mutex since the write lock (in topology.go) is the one we could potentially do more often. And if we don't have any invalid fields, then the lock is only ever acquired once per second, which is nothing

Reply:
Yes, you are right. But I saw that [here](https://github.com/AdRoll/baker/blob/6f870df8405b6f4fbefa5136f357d03773c102e3/stats.go#L171) in the same file the `RLock()` is used to do the same thing. Maybe we could unify the use.

Reply:
yeah there's probably a way to loop only once over the map. But at this point it's more of a cosmetics change since the use is not problematic, anyway it's not the target of this pull-request. I create a ticket on our jira board for the unification